### PR TITLE
add: auth.go test

### PR DESCRIPTION
**Pull request by atlekbai**
- [x] Added test for `GetValidPasswords` in `src/internal/pkg/auth/auth.go`

### CODE REVIEW

**Review by MarvinJWendt — CHANGES_REQUESTED** [review]
Hi, thanks for your PR! I have added some notes. Please fix them and this PR will be ready to merge :)
And please also fix the linter errors. (replace `os.Setenv` with `t.Setenv()` to make the environment change only visible inside the test)

File: src/internal/pkg/auth/auth_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/MarvinJWendt/traefik-guardian/src/internal/pkg/config"
+)
+
+func TestGetValidPasswords(t *testing.T) {
+	os.Setenv("AUTH_HOST", "*")
```
Comment (MarvinJWendt) [inline src/internal/pkg/auth/auth_test.go:12]: Please use `t.Setenv()`

File: src/internal/pkg/auth/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	os.Setenv("PASSWORDS", "plaintext:pass1|pass2|pass3")
```
Comment (MarvinJWendt) [inline src/internal/pkg/auth/auth_test.go:13]: Please use `t.Setenv()`

**Comment by atlekbai** [comment]
Hello! Thank you for the feedback. Fixed according to the review notes.

**Review by MarvinJWendt — APPROVED** [review]
LGTM! Thanks for contributing and happy Hacktoberfest 🚀

**Pull request merged**